### ISSUE TITLE
JVB container configuration question

### ISSUE THREAD

**Issue by blazux** [jitsi/docker-jitsi-meet#534]
Hi,

I've been using the kubernetes deployment of jitsi (which runs flawlessly) and now I'm trying to improve it a bit, I've separated all the containers and I'm trying to connect them the right way at the moment, but I'm unable to configure jvb to connect to the prosody address :

```
env:
 - name: XMPP_SERVER
   value: xmpp.jitsi.svc
```
It's working fine for jifoco : 
```
Jicofo 2020-04-29 13:09:23.609 INFO: [10] org.jitsi.impl.configuration.ConfigurationServiceImpl.log() org.jitsi.jicofo.HOSTNAM
E=xmpp.jitsi.svc
```
But jvb always try to connect to localhost : 
```
WARNING: [MucClient id=shard hostname=localhost] error connecting
org.jivesoftware.smack.SmackException$ConnectionException: The following addresses failed: 'localhost:5222' failed because: localhost/127.0.0.1 exception: java.net.ConnectException: Connection refused (Connection refused)
```

From what I see everything seems fine : 
```
root@jitsi-jvb-74cdbf5c5c-gvbk4:/# env | grep XMPP_SERVER
XMPP_SERVER=xmpp.jitsi.svc
root@jitsi-jvb-74cdbf5c5c-gvbk4:/# telnet xmpp.jitsi.svc 5222
Trying 10.233.1.166...
Connected to xmpp.jitsi.svc.cluster.local.
Escape character is '^]'.

```
Does jvb get the server adress from somewhere else ? What am I doing wrong ?

Thanks a lot !

**Comment by saghul** [comment]
That “localhost” there looks suspicious.

**Comment by blazux** [comment]
Yes it does, but so far I'm unable to spot where it comes from. Is there another variable to configure for this ?

**Comment by saghul** [comment]
Looks like XMPP_SERVER I think. Check the sip-communicator.properties file.

**Comment by nemaniarjun** [comment]
This is interesting! 
I have been trying to setup jisti in fargate-run AWS ECS. 
Fargate runs all the containers (in one task) on the same host and thus does not allow for custom hostnames. Essentially all the services are running on different ports of the same host (localhost) 

Any tips on how to make that work?

**Comment by bajayb4u** [comment]
hi @nemaniarjun: I am trying to setup jitsi-meet in  fargate-run AWS ECS with docker compose file provided in docker repo but I am getting network issues. One of the issue I am currently getting is as below
"ERRO[0000] Error registering task definition             error="ClientException: Fargate only supports network mode ‘awsvpc’"

can you help how to fix this? also if you have list of changes that you have to made to run it on Fargate, can you please share them?

**Comment by ghabarid** [comment]
Hi
Is there any news about that issue?. I m facing the same prob.
Thx

**Comment by goacid** [comment]
https://stackoverflow.com/questions/48809333/how-to-configure-hostname-in-ecs-fargate-task-definition
Help a lot !

**Comment by nemani** [comment]
I did basically what's mentioned in one of the answers. I set all the various jitsi urls to localhost. (in my case I was running all 4 containers in the same task)

If you want to use multiple tasks, then you'll need to setup DNS resolution for them

**Comment by goacid** [comment]
Yes we do run dns resolution. We have at least 3 tasks : 

- 2 for jvb
- 1 for all other

And maybe more for jvb...
